### ISSUE TITLE
Playlist sorting order

### ISSUE THREAD

**Issue by phistep** [Feuermurmel/youtube-podcast-gateway#11]
The playlist [Shore, Stein, Papier, (`PLpr-NGsAGodEbDePSO3wivni39lgdLQjW`](https://www.youtube.com/playlist?list=PLpr-NGsAGodEbDePSO3wivni39lgdLQjW) is sorted in a way that gets the order of episodes right in respect to episode number. I suppose you use the upload date for the `<pubDate>` field in feed generation.  I couldn't really find that in the feed xml, but it appears to me, that several episodes have been published in the same minute. The Pocket Casts podcast player doesn't sort them alphabetically but appearingly random (my guess is it maps to their internal data structure and maybe which item was first presented in the feed and parsed into their data structure or something), which leads to a confused playlist order due to the episodes being published on the same `<pubDate>`. This again is a Pocket Casts problem, but apart from that: Would you be interested in creating an option [like a GET option `?keeporder=true`) for playlists to keep them in order of the playlist creation and not video upload by faking the episode `<pubDate>` starting from 1970-01-01 00:00 and setting each episode 1 hour apart or something like that? This would also be useful for playlists which are created from videos with a broad range of upload date but that is sorted in a special way using the YouTube playlist feature.

**Comment by Feuermurmel** [comment]
It seems that episode order is not kept when retrieving a playlist. It was basically random. I fixed that in https://github.com/Feuermurmel/youtube-podcast-gateway/pull/14. Can you check out https://github.com/Feuermurmel/youtube-podcast-gateway/tree/playlist-order and see if that solves the problem for you?

I looked the episodes of that show and I don't think that the `<pubDate>` values are too near to each other to cause these issues:

```
michi ~$ curl 'http://localhost:8080/playlist/PLpr-NGsAGodEbDePSO3wivni39lgdLQjW' | tidy -xml -i 2> /dev/null | grep '<pubDate>' | sort
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  135k    0  135k    0     0  40256      0 --:--:--  0:00:03 --:--:-- 40283
[...]
      <pubDate>Wed, 19 Aug 2015 17:28:14 -0000</pubDate>
      <pubDate>Wed, 19 Aug 2015 17:29:25 -0000</pubDate>
      <pubDate>Wed, 21 Oct 2015 15:27:41 -0000</pubDate>
      <pubDate>Wed, 21 Oct 2015 15:28:00 -0000</pubDate>
      <pubDate>Wed, 22 Jul 2015 17:44:29 -0000</pubDate>
      <pubDate>Wed, 22 Jul 2015 17:45:22 -0000</pubDate>
      <pubDate>Wed, 22 Jul 2015 17:46:24 -0000</pubDate>
      <pubDate>Wed, 22 Jun 2016 13:18:13 -0000</pubDate>
      <pubDate>Wed, 23 Sep 2015 19:56:52 -0000</pubDate>
      <pubDate>Wed, 23 Sep 2015 19:57:28 -0000</pubDate>
      <pubDate>Wed, 23 Sep 2015 19:58:56 -0000</pubDate>
      <pubDate>Wed, 23 Sep 2015 19:59:56 -0000</pubDate>
      <pubDate>Wed, 24 Jun 2015 16:55:18 -0000</pubDate>
      <pubDate>Wed, 24 Jun 2015 16:55:43 -0000</pubDate>
      <pubDate>Wed, 24 Jun 2015 16:56:32 -0000</pubDate>
[...]
```

**Comment by Feuermurmel** [comment]
I just merged #14. I'm closing this issue.

Please re-open if the issue persists.
